keps/sig-node/5304-dra-attributes-downward-api/README.md

```diff
@@ -0,0 +1,830 @@
+
+The Kubernetes Downward API provides a standard mechanism for exposing Pod and container metadata to workloads. Extending this API to support DRA device attributes would enable workloads to discover device information without requiring additional custom controllers or privileged access to the Kubernetes API.
+
+### Goals
+
+- Provide a stable Downward API path for device attributes associated with `pod.spec.resourceClaims[*]` requests
+- Support device attributes from `ResourceSlice` that are requested by user in pod spec
+- Maintain compatibility with existing DRA drivers without requiring changes to driver interfaces
+
+### Non-Goals
+
+- Expose the entirety of `ResourceClaim`/`ResourceSlice` objects in the Downward API
+- Allow arbitrary JSONPath into external objects via Downward API
+- Change or extend DRA driver interfaces
+- Support dynamic updates to device attributes after Pod container startup (for env vars)
+ - Propagate or snapshot attributes from `ResourceSlice` into `ResourceClaim` at allocation time (no scheduler involvement)
+
+## Proposal
+
+This proposal introduces a new Downward API selector (`resourceSliceAttributeRef`) that allows Pods to reference DRA device attributes in environment variables. The kubelet will implement a local DRA attributes controller that:
+
+1. Watches Pods scheduled to the node and identifies those with `pod.spec.resourceClaims`
+2. Watches `ResourceClaim` objects in the Pod's namespace to retrieve allocation information
+3. Watches `ResourceSlice` objects for the node and driver to resolve device attributes
+4. Maintains a per-Pod cache of `(claimName, requestName) -> {attribute: value}` mappings
+5. Resolves `resourceSliceAttributeRef` references when containers start
+
+Downward API references expose one attribute per reference. The kubelet resolves only the attribute explicitly referenced via `ResourceSliceAttributeSelector`.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a KubeVirt developer, I want the virt-launcher Pod to automatically discover the PCIe root address of an allocated physical GPU via environment variables, so that it can construct the libvirt domain XML to pass through the device to the virtual machine guest without requiring a custom controller.
+
+#### Story 2
+
+As a DRA driver author, I want my driver to remain unchanged while allowing applications to consume device attributes (like `resource.kubernetes.io/pcieRoot` or `dra.kubervirt.io/mdevUUID`) through the native Kubernetes Downward API.
+
+### Notes/Constraints/Caveats (Optional)
+
+- Environment variables are set at container start time: Once a container starts, its environment variables are immutable. If device attributes change after container start, env vars will not reflect the change.
+- Resolution timing: Attributes are resolved at container start time (not at allocation time). There is no scheduler-side copying of attributes into `ResourceClaim`.
```

Comment:
please note the kubelet restart or container crash scenarios. The behavior must be declared in those cases. Ideally something aligned with https://github.com/kubernetes/enhancements/issues/3721

We need to clearly articulate that the crash/restart of a container may lead to it's unavilability to start if attribute or resource has disappeared